### PR TITLE
(CDAP-18000) Upgrade k8s client to latest

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java [hunk 1]
```diff
@@ -166,14 +167,12 @@ public ResourceReport getResourceReport() {
...
+      api.deleteNamespacedPodAsync(podName, kubeNamespace, null, null, null, null, null, deleteOptions,
+                                   createCallbackFutureAdapter(resultFuture, r -> runnable));
     } catch (ApiException e) {
       completeExceptionally(resultFuture, e);
-    } catch (Exception ex) {
-      // TODO: CDAP-18000 Ignore any exception apart from ApiException till we upgrade the kubernetes client library
-      LOG.trace("Ignoring exception while deleting pod {}", podName, ex);
     }
+
     resultFuture.complete("success");
```
Comment (sagarkapare) [inline cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java]: This need to be removed too.

File: cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java [hunk 2]
```diff
@@ -596,7 +596,8 @@ private V1ObjectMeta createDeployment(V1ObjectMeta metadata,
 
     V1Deployment deployment = buildDeployment(metadata, runtimeSpecs, runtimeConfigLocation);
 
-    deployment = appsApi.createNamespacedDeployment(kubeNamespace, deployment, "true", null, null);
+    deployment = appsApi.createNamespacedDeployment(kubeNamespace, deployment, "true", null
+      , null);
```
Comment (sagarkapare) [inline cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java]: nit: indentation seems off.

**Review by sagarkapare — APPROVED** [review]
couple of comments, else LGTM 👍

File: cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java (hunk 1 above)
Comment (chtyim) [reply cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java]: Good catch

File: cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java (hunk 2 above)
Comment (chtyim) [reply cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java]: fixed

**Review by CuriousVini — APPROVED** [review]
lgtm 👍🏼

**Comment by CuriousVini** [comment]
build is green merging to rebase #13525

**Pull request merged**